lib/application.js

```diff
@@ -0,0 +1,186 @@
+var url = require("url");
+var qs = require("qs");
+
+/**
+ * Application
+ * @constructor
+ * @param {Object} client Catapult client
+ */
+
+var Application = function (client) {
+	/**
+	 * List the user's applications
+	 * @param {Object} params Parameters for filtering applications
+	 * @param {Number} [params.size] The size of each page requested (Max size: 1000)
+	 * @param {Function} [callback] A callback for the list of applications
+	 * @return {MessageResponse} A promise for the list of applications
+	 */
+	this.list = function (params, callback) {
+		var self = this;
+		return client.makeRequest({
+			path   : "applications",
+			method : "GET",
+			qs     : params
+		})
+		.then(function (response) {
+			var applicationListResponse = {
+				applications : response.body
+			};
+			var links = response.headers.link.split(",");
+			var nextLink = links[1];
```

Comment:
It is not guaranteed that the next link will always be the 2nd link, also we will need this code in several places so we should probably make a function that grabs the next link. Some resources, such as message don't contain the 'first' link so you will actually need to check if the link matches.

I have some example code that does this in the Rust SDK if that helps: https://github.com/bandwidthcom/rust-bandwidth/blob/master/src/util.rs#L15